src/sage/groups/perm_gps/partn_ref/data_structures.pxd

```diff
@@ -136,6 +136,37 @@ cdef inline void OP_join(OrbitPartition *OP, int m, int n) noexcept:
     if m_root != n_root:
         OP.num_cells -= 1
 
+
+cdef inline void OP_make_set(OrbitPartition *OP) noexcept:
+    cdef int i, n = OP.degree
+    cdef int *new_parent, *new_rank, *new_mcr, *new_size
+
+    cdef int *int_array = <int *> sig_malloc( 4*(n+1) * sizeof(int) )
+    if int_array is NULL:
+        sig_free(int_array)
+    else :
+        OP.degree = n + 1
+        OP.num_cells = OP.num_cells + 1
+        new_parent = int_array
+        new_rank = int_array + (n + 1)
+        new_mcr = int_array + (2*n + 2)
+        new_size = int_array + (3 * n + 3)
+        for i from 0 <= i < n:
+            new_parent[i] = OP.parent[i]
+            new_rank[i] = OP.rank[i]
```

Comment:
why not using low level `memcpy` method ?

Reply:
I'm not perfect with c, but if I understand correctly, `memcpy` will also copy the amount of memory allocated for the object. Since we are working with arrays, that would mean that the new copy would not have enough memory allocated to it since we need to add an additional entry into the array.

Reply:
The first option is to make 4 different mallocs for parent, rank, etc. when first creating the data structure.
Then, in `make_set`, you can use `realloc` to extend the size of the arrays. Realloc should copy the data for you and so you only have to allocate the last value. In case of error, one of these arrays will get a NULL pointer (unless I'm mistaken). Last, when deallocating, you have to do 4 free.

The other option is to use `memcpy` instead of the for loop to put the good values in `new_...`. It should be a call like `memcpy(new_parent, OP.parent, n)`. It remains to assign the last cell of `new_parent`.

Reply:
Ok, I'm not as comfortable with C, so I'll see if I can try and use these methods.